transmogrifier/sources/xml/datacite.py

```diff
@@ -425,6 +363,100 @@ def get_locations(cls, source_record: Tag) -> list[timdex.Location] | None:
             )
         ] or None
 
+    @classmethod
+    def get_notes(cls, source_record: Tag) -> list[timdex.Note] | None:
+        notes = []
+        notes.extend(list(cls._get_resource_type_note(source_record)))
+        notes.extend(list(cls._get_description_notes(source_record)))
+        return notes or None
+
+    @classmethod
+    def _get_resource_type_note(cls, source_record: Tag) -> Iterator[timdex.Note]:
+        if resource_type := source_record.metadata.find("resourceType", string=True):
+            yield timdex.Note(
+                value=[str(resource_type.string)],
+                kind="Datacite resource type",
+            )
+
+    @classmethod
+    def _get_description_notes(cls, source_record: Tag) -> Iterator[timdex.Note]:
+        descriptions = source_record.metadata.find_all("description", string=True)
+        for description in descriptions:
+            if "descriptionType" not in description.attrs:
+                logger.warning(
+                    "Datacite record %s missing required Datacite attribute "
+                    "@descriptionType",
+                    cls.get_source_record_id(source_record),
+                )
+            if description.get("descriptionType") != "Abstract":
+                yield timdex.Note(
+                    value=[str(description.string)],
+                    kind=description.get("descriptionType") or None,
+                )
+
+    @classmethod
+    def get_publishers(cls, source_record: Tag) -> list[timdex.Publisher] | None:
+        publishers = []
+        if publisher := source_record.metadata.find("publisher", string=True):
+            publishers.append(timdex.Publisher(name=str(publisher.string)))
+        else:
+            logger.warning(
+                "Datacite record %s missing required Datacite field publisher",
+                cls.get_source_record_id(source_record),
+            )
+        return publishers or None
```

Comment:
Good point, I was going for consistency but this is more logically sound. I also updated `get_content_types` and `get_languages` since they used the same unnecessary list pattern